common/quotas/clocked_rate_limiter_test.go

```diff
@@ -151,3 +151,65 @@ func TestClockedRateLimiter_Wait_DeadlineWouldExceed(t *testing.T) {
 	t.Cleanup(cancel)
 	assert.ErrorIs(t, rl.Wait(ctx), quotas.ErrRateLimiterReservationWouldExceedContextDeadline)
 }
+
+// test that reservations for 1 token ARE unblocked by RecycleToken
+func TestClockedRateLimiter_Wait_Recycle(t *testing.T) {
+	t.Skip("flaky test")
+	t.Parallel()
+	ts := clock.NewEventTimeSource()
+	rl := quotas.NewClockedRateLimiter(rate.NewLimiter(1, 1), ts)
+	ctx := context.Background()
+
+	// take first token
+	assert.NoError(t, rl.Wait(ctx))
+
+	// wait for next token and report when success
+	asserted := false
+	go func() {
+		assert.NoError(t, rl.Wait(ctx))
+		asserted = true
+	}()
+	// wait for rl.Wait() to start and get to the select statement
+	time.Sleep(time.Second) // nolint
```

Comment:
This doesn't seem right.. at the start of the second there are no tokens, but after a whole second, there should be 1 (since the rate is 1/sec). so this Wait could return either naturally or due to the Recycle, and we can't tell the difference. It should more like sleep 10ms to let it get blocked, then recycle, then check that Wait returned within 10ms after that (20ms total), much less than 1s (the natural wakeup)

Reply:
Yeah, the test wasn't passing in CI due to detecting a race, so I tried the 1s sleep just to see if it would help the test not have a race. But that didn't even work :/

Reply:
reverted to 10ms